js/src/forum/components/Search.js

```diff
@@ -158,7 +155,7 @@ export default class Search extends Component {
 
         clearTimeout(search.searchTimeout);
         search.searchTimeout = setTimeout(() => {
-          if (search.searched.indexOf(query) !== -1) return;
+          if (state.cachedSearches.indexOf(query) !== -1) return;
```

Comment:
Now that we're no longer operating on `this`, but a dependency instead, it might be worth encapsulating this logic in some nicely named methods? E.g. `state.isCached()` and (for the next change) `state.cache()`.

Reply:
yeah I'd agree with that.